app/jobs/suitable_for_early_allocation_email_job.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+class SuitableForEarlyAllocationEmailJob < ApplicationJob
+  queue_as :mailer
+
+  EQUIP_URL = 'https://equip-portal.rocstac.com/CtrlWebIsapi.dll/?__id=webDiagram.show&map=0%3A9A63E167DE4B400EA07F81A9271E1944&dgm=4F984B45CBC447B1A304B2FFECABB777'
+
+  def perform
+    offenders = EarlyAllocation.suitable_offenders_pre_referral_window
+
+    offenders.each do |offender_no|
```

Comment:
I'm wondering whether a more resilient approach would be to schedule one job per offender.

I'm thinking the job itself could effectively be whatever's inside this `offenders.each` block, with the job accepting `offender_no` as an argument. And then we could move the call to `EarlyAllocation.suitable_offenders_pre_referral_window` into the rake task, and have it schedule one job for each offender.

My reason for preferring one job per offender, rather than one job for all offenders, is that we can avoid one bad offender stopping the other emails from going out. For example, if one offender mid-way through the list is broken – e.g. either the Prison API or GOV.UK Notify raises an error, and it means we can't send the email – the job will fail only for that one offender. All other offenders are unaffected because they're done in independent jobs. And any failed offenders will be retried independently, rather than needing to loop through them all again.

Reply:
I don't disagree with this